fml/thread.h

```diff
@@ -16,7 +17,52 @@ namespace fml {
 
 class Thread {
  public:
-  explicit Thread(const std::string& name = "");
+  // Valid values for priority of Thread
+  enum class ThreadPriority : int {
+    // Suitable for threads that shouldn't disrupt high priority work.
+    BACKGROUND,
+    // Default priority level.
+    NORMAL,
+    // Suitable for threads which generate data for the display.
+    DISPLAY,
+    // Suitable for thread which raster data
+    RASTER,
+  };
+
+  /// The ThreadConfig is used for setting thread perorities
+  class ThreadConfig {
+   public:
+    explicit ThreadConfig(std::string name = "",
+                          ThreadPriority priority = ThreadPriority::NORMAL);
+
+    static std::unique_ptr<ThreadConfig> DefaultConfigure(
+        const std::string& name = "") {
+      return std::make_unique<ThreadConfig>(name);
+    }
+
+    ThreadPriority thread_priority() const { return thread_priority_; }
+
+    const std::string& thread_name() const { return thread_name_; }
+
+    /// Set current thread name
+    virtual void SetCurrentThreadName();
```

Comment:
Why is this virtual?  Maybe instead of making this a class, you want it to be an interface?

Reply:
Yes, I think the method in the ThreadConfigure class user can have their own implementation by the user.